x-pack/plugins/security/server/authentication/authentication_service.ts

```diff
@@ -0,0 +1,207 @@
+  ILegacyClusterClient,
+  HttpServiceStart,
+} from '../../../../../src/core/server';
+import type { SecurityLicense } from '../../common/licensing';
+import type { AuthenticatedUser } from '../../common/model';
+import type { AuditServiceSetup, SecurityAuditLogger } from '../audit';
+import type { ConfigType } from '../config';
+import type { SecurityFeatureUsageServiceStart } from '../feature_usage';
+import type { Session } from '../session_management';
+import type { DeauthenticationResult } from './deauthentication_result';
+import type { AuthenticationResult } from './authentication_result';
+import { getErrorStatusCode } from '../errors';
+import { APIKeys } from './api_keys';
+import { Authenticator, ProviderLoginAttempt } from './authenticator';
+
+interface AuthenticationServiceSetupParams {
+  legacyAuditLogger: SecurityAuditLogger;
+  audit: AuditServiceSetup;
+  getFeatureUsageService: () => SecurityFeatureUsageServiceStart;
+  http: HttpServiceSetup;
+  clusterClient: ILegacyClusterClient;
+  config: ConfigType;
+  license: SecurityLicense;
+  loggers: LoggerFactory;
+  session: PublicMethodsOf<Session>;
+}
+
+interface AuthenticationServiceStartParams {
+  http: HttpServiceStart;
+  clusterClient: IClusterClient;
+}
+
+export interface AuthenticationServiceSetup {
+  /**
+   * @deprecated
+   */
+  getCurrentUser: (request: KibanaRequest) => AuthenticatedUser | null;
+}
+
+export interface AuthenticationServiceStart {
+  apiKeys: Pick<
+    APIKeys,
+    | 'areAPIKeysEnabled'
+    | 'create'
+    | 'invalidate'
+    | 'grantAsInternalUser'
+    | 'invalidateAsInternalUser'
+  >;
+  login: (request: KibanaRequest, attempt: ProviderLoginAttempt) => Promise<AuthenticationResult>;
+  logout: (request: KibanaRequest) => Promise<DeauthenticationResult>;
+  acknowledgeAccessAgreement: (request: KibanaRequest) => Promise<void>;
+  getCurrentUser: (request: KibanaRequest) => AuthenticatedUser | null;
+}
+
+export class AuthenticationService {
+  private license!: SecurityLicense;
+  private authenticator!: Authenticator;
+
+  constructor(private readonly logger: Logger) {}
+
+  setup({
+    legacyAuditLogger: auditLogger,
+    audit,
+    getFeatureUsageService,
+    http,
+    clusterClient,
+    config,
+    license,
+    loggers,
+    session,
+  }: AuthenticationServiceSetupParams): AuthenticationServiceSetup {
+    this.license = license;
+
+    const getCurrentUser = (request: KibanaRequest) => {
+      if (!license.isEnabled()) {
+        return null;
+      }
+
+      return (http.auth.get(request).state ?? null) as AuthenticatedUser | null;
+    };
+
+    this.authenticator = new Authenticator({
+      legacyAuditLogger: auditLogger,
+      audit,
+      loggers,
+      clusterClient,
+      basePath: http.basePath,
+      config: { authc: config.authc },
+      getCurrentUser,
+      getFeatureUsageService,
+      license,
+      session,
+    });
+
+    this.logger.debug('Successfully initialized authenticator.');
```

Comment:
**question** Is this log statement necessary? If the authenticator doesn't initialize, will Kibana even be able to start correctly?

Reply:
Good catch, I think it's useless, will remove it.

>  If the authenticator doesn't initialize, will Kibana even be able to start correctly?

I don't think so